### ISSUE TITLE
Unable to concatenate strings while looping a list using with_items in Ansible 2.5.1

### ISSUE THREAD

**Issue by maduhshan** [ansible/ansible#39181]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
with_items

##### ANSIBLE VERSION
```
ansible 2.5.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ijiffry/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)
```

##### CONFIGURATION
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles']
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False

##### OS / ENVIRONMENT
RHEL6

##### SUMMARY
A string cannot be concatenated via a loop. Finally appended value is displayed as a result.

##### STEPS TO REPRODUCE

```yaml
---
- hosts: 127.0.0.1
  connection: local

  vars:
    tmpvar: ""

   // some lines here.... 

   # appending available fields
    - name: appending available fields
      block:
      - set_fact:
          tmpvar: "{{tmpvar}}-{{item.split('=')[0]}}"
        when: item.split('=')[0] in required_fields[endpoint] + credential_fields and item.split('=')[1] | trim != ""
        with_items: "{{query_items}}"
      - set_fact:
          available_fields: "{{tmpvar.split('-')}}"
      - set_fact:
          error_msgs: "{{error_msgs}} + ['{{item}} not specified or value cannot be empty']"
        when: item not in available_fields
        with_items: "{{required_fields[endpoint] + credential_fields}}"
      when: endpoint is defined and lastline_endpoint | trim != ""
```

##### EXPECTED RESULTS
var tmpvar: "test"
//appending another string to existing var
test-sometxt-somemore-finaltxt

##### ACTUAL RESULTS
// but only displayed the finally appended value
-finaltxt

**Comment by sivel** [comment]
This is resolved in https://github.com/ansible/ansible/pull/38302 and will be included in 2.5.2

**Comment by maduhshan** [comment]
https://github.com/ansible/ansible/releases 

Ansible 2.5.2 is not released yet. Any work around within 2.5.1?
